### PR TITLE
Final Model testing for Rohan Himal

### PR DESCRIPTION
**Pull request by anonymous-himalayas**
Himal:
Base model of Yolo26m using 75 epochs

Utilized a combination of undersampling and oversampling techniques along with a 10% dataset ablation (didn't end up doing 15% since I was already losing some information from undersampling). Did some hyperparameter tuning by changing a couple of base values such as mosaic_plot and cls. Decided to go with a combination of undersampling and oversampling since it would allow all classes to have the same number of images rather than some having more and some less. 

Things I could have done given more resources:
More epochs, bigger model (Yolo26l or Yolo26x)
Possibly 15% dataset ablation just to test it out
Final Testing accuracy:
Test Results:
mAP@0.5: 0.7663
mAP@0.5:0.95: 0.6121
Precision: 0.7433
Recall: 0.6803

Rohan:
Base model of YOLO26x with 58 epochs

Performed dataset ablation on V2 on the bottom 15% of each class in terms of quality aka laplacian variance (Link to new dataset creation: https://colab.research.google.com/drive/1R1bQTN40hVIm6bjtMkoiA8axoeyY_7ga?usp=sharing).
The new dataset has been uploaded to the CareTech workspace in Roboflow which is already shared with everybody and is under the project id 'final_training_w_v2_ablation'.

Preprocessing - Auto-orient, 640x640 resizing
Augmentations - None

I honestly wasn't too impressed with this model. I feel that I've been able to obtain similar results with the lighter v11, v26 versions, and using an XL model didn't make it as accurate as I thought it would. I did run out of credits and it seemed like the model was continuing to progress, however, which makes any evaluations I give a bit inaccurate. I know there's not much more time, and credits are dwindling, but I would love to fully train a huge model like this one and see how good it can actually get.
Testing Results
mAP@50 77.3%
Precision 79.4%
Recall 70.1%
F1 72.9%

### CODE REVIEW

**Review by margoglvz — CHANGES_REQUESTED** [review]
Hi @rohanb123 and @anonymous-himalayas!

Thank you so much for your PR and for all your hard work!

Himal: Thanks for using the YOLO specific hyperparameters as that hasn't been tried out before! I saw that you chose mosaic_plot and cls for your hyperparameters to change and was wondering why you chose those specifically? Also, super nice how you used a combination of oversampling and undersampling to make sure classes were equal and I liked how organized the code was, made it easy to read. I also see that you used only the V1 for the model, I was just curious as to how you think the model would perform if you used V2?

Rohan: Thank you for trying out the 26x and really cool that you used no augmentations! Also, can you provide a link to the 'final_training_w_v2_ablation' workspace? I checked both CareTech workspaces and wasn't able to find it 😢. I think it is also really nice it was able to achieve such a high performance with 58 epochs as I would have expected it would have needed above 70. Also just to double check, was the meatloaf class dropped LOL

Great job to both of you for getting your results above ~70% for mAP, precision and recall! When Leo and I get more credits and more time, we will def look back at your guys' modifications and keep it in mind if we do any further changes.

When you get a chance, def share the link to the workspace cause for some reason I can't see it 😭  just to make sure I can review everything. But thank you so much again and I hope you guys have a great spring break!
